src/diffusers/models/transformers/hunyuan_transformer_2d.py

```diff
@@ -12,129 +12,237 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from dataclasses import dataclass
-from typing import Any, Dict, Optional
+from typing import Optional
 
 import torch
 import torch.nn.functional as F
 from torch import nn
 
 from ...configuration_utils import ConfigMixin, register_to_config
 from ...utils import BaseOutput, deprecate, is_torch_version, logging
-from ..attention import HunyuanDiTBlock, FP32_SiLU, FP32_Layernorm, HunyuanDiTAttentionPool, modulate
+from ...utils.torch_utils import maybe_allow_in_graph
+from ..attention import FeedForward
+from ..attention_processor import Attention, HunyuanAttnProcessor2_0
 from ..modeling_utils import ModelMixin
-from ..normalization import AdaLayerNormSingle
+from ..normalization import AdaLayerNormSingle, AdaLayerNormContinuous
 from .transformer_2d import Transformer2DModelOutput
+from ..embeddings import Timesteps, TimestepEmbedding, PatchEmbed, HunYuanTextProjection, get_timestep_embedding
 
 
 logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
 
 import math
 from einops import repeat
 
-class HunyuanDiTPatchEmbed(nn.Module):
-    """ 2D Image to Patch Embedding
+class FP32_Layernorm(nn.LayerNorm):
+    def forward(self, inputs: torch.Tensor) -> torch.Tensor:
+        origin_dtype = inputs.dtype
+        return F.layer_norm(inputs.float(), self.normalized_shape, self.weight.float(), self.bias.float(),
+                            self.eps).to(origin_dtype)
 
-    Image to Patch Embedding using Conv2d
 
-    A convolution based approach to patchifying a 2D image w/ embedding projection.
+class AdaLayerNormShift(nn.Module):
```

Comment:
This is refactored from `default_modulation`: can you tell us more about it? Is it part of the `AdaLayerNorm` family but without the scale?  is there a reference? should we move to `embeddings.py` with other normalization methods?